app/react/hierarchy/keys/Key.js

```diff
@@ -11,48 +11,89 @@ type PropTypes = {
   location: LocationWithQuery,
   childrenKeys: Array<Object>,
   style: Object,
+  collapsed: boolean,
   createStyles: (level: number) => Object,
   path: Array<string>,
-  setPath: Function
+  setPath: Function,
+  isCollapsed: (Array<string>, Array<string>) => boolean,
+  globalPath: Array<string>
 };
 
-export default function Key({
-  dispatch,
-  label,
-  childrenKeys,
-  location,
-  style,
-  createStyles,
-  path,
-  setPath
-}: PropTypes) {
-  const currentPath = [...path, label];
-  return (
-    <div style={style} >
-      <LabelLink
-        path={currentPath}
-        location={location}
-        label={label}
-      />
-      <div>
-        {childrenKeys.map(key => (
-          <Key
-            dispatch={dispatch}
-            key={key.label}
-            setPath={setPath}
-            style={createStyles(key.level)}
-            createStyles={createStyles}
-            label={key.label}
-            path={currentPath}
-            location={location}
-            childrenKeys={key.childrenKeys}
-          />
-        ))}
+type StateTypes = {
+  collapsed: boolean
+};
+
+export default class Key extends React.Component {
+  static defaultProps = {
+    path: []
+  }
+
+  constructor(props: PropTypes) {
+    super(props);
+    this.state = { collapsed: props.collapsed };
+  }
+
+  state: StateTypes
+
+  componentWillReceiveProps = ({ collapsed }) => {
+    if (collapsed !== this.state.collapsed) this.setState({ collapsed });
+  }
+
+  props: PropTypes
+
+
+  toggle = () =>
+    this.setState({ collapsed: !this.state.collapsed })
+
+  render() {
+    const { collapsed } = this.state;
+    const {
+      dispatch,
+      label,
+      childrenKeys,
+      location,
+      style,
+      createStyles,
+      path,
+      setPath,
+      globalPath,
+      isCollapsed
+    } = this.props;
+    const currentPath = [...path, label];
+
+    return (
+      <div style={style}>
+        <button onClick={this.toggle}>
+          {/* Very high tech solution. I'm available to talk about this on TED or Techcrunch. */}
+          {collapsed ? 'V' : '>'}
```

Comment:
It will be better looking if you use of HTML notation of proper UTF-8 symbol.
https://www.w3schools.com/charsets/ref_utf_geometric.asp